database/app/src/main/java/com/google/firebase/referencecode/database/MainActivity.java

```diff
@@ -29,32 +30,35 @@
 public class MainActivity extends AppCompatActivity implements MainActivityInterface {
 
     private static final String TAG = "MainActivity";
+    DatabaseReference myRef;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        FirebaseDatabase database = FirebaseDatabase.getInstance();
+        myRef = database.getReference("message");
     }
 
     @Override
-    public void basicReadWrite() {
-        // [START write_message]
-        // Write a message to the database
-        FirebaseDatabase database = FirebaseDatabase.getInstance();
```

Comment:
We actually do want to show the initialization of the `database` and `myRef` variables as part of the snippet here.  For many people this code snippet is the first Firebase code they ever see and we want to be very explicit.